WordPress/src/main/java/org/wordpress/android/ui/media/MediaPreviewActivity.java

```diff
@@ -345,28 +414,40 @@ public void onMediaTapped() {
 
         @Override
         public Fragment getItem(int position) {
-            int id = Integer.valueOf(mMediaIdList.get(position));
-            MediaModel media = mMediaStore.getMediaWithLocalId(id);
-
-            // make sure we autoplay the initial item (relevant only for audio/video)
-            boolean autoPlay;
-            if (id == mMediaId && !mDidAutoPlay) {
-                autoPlay = true;
-                mDidAutoPlay = true;
-            } else {
-                autoPlay = false;
+            MediaPreviewFragment fragment;
+            switch (getPreviewType()) {
+                case MULTI_MEDIA_IDS:
+                    int id = Integer.valueOf(mMediaIdOrUrlList.get(position));
+                    MediaModel media = mMediaStore.getMediaWithLocalId(id);
+                    // make sure we autoplay the initial item (relevant only for audio/video)
+                    boolean autoPlay;
+                    if (id == mMediaId && !mDidAutoPlay) {
+                        autoPlay = true;
+                        mDidAutoPlay = true;
+                    } else {
+                        autoPlay = false;
+                    }
+                    fragment = MediaPreviewFragment.newInstance(mSite, media, autoPlay);
+                    break;
+                case MULTI_IMAGE_URLS:
+                    String imageUrl = mMediaIdOrUrlList.get(position);
+                    fragment = MediaPreviewFragment.newInstance(null, imageUrl);
+                    break;
+                default:
+                    // should never get here
+                    return null;
```

Comment:
Minor: If I understand this correctly, the default case here is defined just as a shortcut to bail with null and please the compiler for the "variable may not have been initialized" error on linenumber 441. If so, and since this is case that should never happen (as the comment denotes), what do you think about throwing here a `RuntimeException("Internal incosistency")` to help us catch it more easily via Crashlytics?

Reply:
Thanks for the review @hypest! I added the exception in 3f5e1fd.